index.js

```diff
@@ -131,10 +131,84 @@ newSightingForm.addEventListener('submit', function (event) {
     renderAnimalSightingPost(newSightingObject)
     newSightingForm.reset()
 
+    
 })
 
 
 /***** End of Starter Code *****/
 /************************** EVENTS PART 2 JS MINI CHALLENGE ******************************/
 
+/***** Deliverable 1 *****/
+
+// document.querySelector('ul#animals').addEventListener('click', function(event){
+//     if (event.target.matches)
+//     console.log('Like clicked!!!', event.target)
+
+// })
+
+/***** Deliverable 2 *****/
+
+// document.querySelector('button.delete-button').addEventListener('click', function(event){
+
+//     console.log('Delete clicked!!!', event.target)
+
+// })
+
+const ulAnimals = document.querySelector('ul#animals')
+
+    // console.log(ulAnimals)
+
+ulAnimals.addEventListener('click', function(event) {
+
+    // console.log(event.target)
+    if(event.target.matches('button.like-button')) {
+        // console.log('like clicked')
+        
+        const likesPtag = event.target.previousElementSibling
+        // console.log(likesPtag)
+        const likes = parseInt(likesPtag.textContent) + 1
+        likesPtag.textContent = `${likes} Likes`
+        // console.log(likesPtag)
+    }
+    else if (event.target.matches('button.delete-button')) {
+        // console.log('delete clicked')
+        const li = event.target.parentElement
+        // console.log(li)
+        li.remove()
+    }
+
+    else if (event.target.matches('button.toggle-update-form-button')) {
+        // console.log('clicked')
+        
+        const sighting = event.target.nextElementSibling
+        if (sighting.style.display === 'block') {
+            sighting.style.display = 'none'
+        }
+        else {
+            sighting.style.display = 'block'
+        }
+
+        // +ternary op!
+        // sighting.style.display = sighting.style.display === 'block' ? 'none' : 'block'
+
+    }
+})
```

Comment:
Nice! You have an event listener on the entire animals UL element, and based on the class names you are delegating out different logic. Great. First step is getting everything up and running. Awesome. Next, we can refactor for readability and to make our code more reusable.

```
/***** DELIVERABLE 1, 2, & 3 part 1 *****/

const animalsUL = document.querySelector('ul#animals')
animalsUL.addEventListener("click", function(event){
    if (event.target.matches('button.like-button')){
        handleLike(event)
    } else if (event.target.matches('button.delete-button')){
        handleDelete(event)
    } else if (event.target.matches('button.toggle-update-form-button')){
        handleToggleUpdate(event)
    }
})

/***** DELIVERABLE 1 part 2 *****/
function handleLike(e){
    const likesDisplayPTag = e.target.previousElementSibling
    const currentTextContent = likesDisplayPTag.innerText
    let numberOfLikes = parseInt(currentTextContent.split(" ")[0])
    numberOfLikes += 1
    const newTextContent = `${numberOfLikes} Likes`
    likesDisplayPTag.innerText = newTextContent
}

/***** DELIVERABLE 2 part 2 *****/
function handleDelete(e){
    const parentLI = e.target.parentElement
    parentLI.remove()
}

/***** DELIVERABLE 3 part 2 *****/
function handleToggleUpdate(e){
    const updateForm = e.target.nextElementSibling
    toggleHidden(updateForm)
}

/***** DELIVERABLE 3 part 3 *****/
function toggleHidden(element){
    element.style.display === "none" ? element.style.display = "block" : element.style.display = "none"
}
```